src/py21cmfast/wrapper.py

```diff
@@ -1902,6 +1890,35 @@ def run_coeval(
         redshifts += redshift
         redshifts = sorted(set(redshifts), reverse=True)
 
+        if (
+            flag_options.PHOTON_CONS
+            and np.amin(redshifts) < global_params.PhotonConsEndCalibz
+        ):
+            raise ValueError(
+                """
+                The final (lowest) redshift (z = %g) is lower than the endpoint of the photon
```

Comment:
I think I would change this particular error message to start with "You have passed a redshift (z = %g) that is lower than...".

Actually for both this and the other error message, you could take advantage of f-strings to put the parameters in, so:

```python
f"""You have passed a redshift (z = {np.amin(redshifts)}) that is lower than..."""
```